src/main/java/de/dhbw/karlsruhe/services/MenuDialogService.java

```diff
@@ -0,0 +1,72 @@
+package de.dhbw.karlsruhe.services;
+
+import java.util.InputMismatchException;
+import java.util.Scanner;
+
+public class MenuDialogService {
+    private Scanner scanner;
+    private String[] menuOptions;
+    private int userInput;
+    private PlayDialogService playDialogService;
+    private LeaderboardDialogService leaderboardDialogService;
+    private LogoutService logoutService;
```

Comment:
Die Service Klassen werden nur in der Methode checkUserInput genutzt, wäre es da sinnvoll, diese erst dort zu erzeugen und den Overhead zu reduzieren?

Reply:
Habe gedacht, dass dies eine "schönere" und übersichtliche Implementierung ist. Weißt du ob es Auswirkungen auf die Performance hat?

Reply:
Momentan: Lebenszeit der Variablen = Lebenszeit der Instanz der Klasse MenuDialogService.
@whole01 Ansatz: Variablen würden nur für den Funktionsaufruf angelegt werden und danach wieder entfernt.